src/blockchain/Blockchain.ts

```diff
@@ -43,12 +43,11 @@ export interface IBlockchainApi {
 // https://polkadot.js.org/api/api/classes/_promise_index_.apipromise.html
 
 export default class Blockchain implements IBlockchainApi {
-  public static asArray(queryResult: QueryResult): AnyJson[] {
-    const json =
-      queryResult && queryResult.encodedLength ? queryResult.toJSON() : null
-    if (json instanceof Array) {
-      return json
-    }
+  public static asArray(queryResult: Codec): AnyJson[] {
+    if (queryResult instanceof Array) return decodeArray(queryResult)
+    // new but in the spirit of the method's name: Struct to Array (drops keys)
+    if (queryResult instanceof Map)
+      return decodeArray(Array.from(queryResult.values()))
     return []
   }
```

Comment:
This is no longer used internally but as it was part of the public API I kept it. A decoder method like this seemed not to belong here, so I moved functionality out to the new Decode module. The behavior changes a little: We now not only decode, but also check for non-emptiness first. Otherwise this method would only do what .toJSON() already does.